### PR TITLE
fix: missed updates for tensorflow 2.11.0

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

These selective imports of (legacy) optimizers were missed in #6527.

## Test Plan

Selected jobs in `test-e2e-longrunning` should pass - this will test the selective import logic, _not_ the legacy optimizer itself (TF 2.8 will be used). The legacy optimizer has already been successfully tested in #6595 (which should rebase after this PR is merged).

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: examples/computer_vision/unets_tf_keras/model_def.py [hunk 1]
```diff
@@ -104,7 +105,11 @@ def build_model(self):
         model = self.context.wrap_model(model)
 
         # Create and wrap optimizer.
-        optimizer = tf.keras.optimizers.Adam()
+        # TODO MLG-443 Migrate from legacy Keras optimizers
+        if version.parse(tf.__version__) >= version.parse("2.11.0"):
+            optimizer = tf.keras.optimizers.legacy.Adam()
+        else:
+            optimizer = tf.keras.optimizers.Adam()
```
Comment (mpkouznetsov) [inline examples/computer_vision/unets_tf_keras/model_def.py:112]: Whereas I agree that for fixtures in e2e tests having this conditionality is the way to go, I am not sure it is appropriate here.

**Comment by drh-determined-ai** [comment]
Do not merge. It was decided that these conditional imports will be unnecessary after an imminent bumpenv. Similar changes made in #6527 will be undone then as well.
